spec/fieldhand/repository_spec.rb

```diff
@@ -124,6 +124,24 @@ module Fieldhand
 
         repository.records(:from => '2001-01-01', :until => '2002-01-01')
       end
+
+      it 'returns all records for repository with namespace' do
+        stub_oai_request('http://www.example.com/oai?verb=ListRecords&metadataPrefix=marcxml', 'list_records_with_namespaces.xml')
+        repository = described_class.new('http://www.example.com/oai')
+
+        records = repository.records(:metadata_prefix => 'marcxml').to_a
+        expect(records.size).to eq(2)
+      end
+
+      it 'populates records with the right information for repository with namespace' do
+        stub_oai_request('http://www.example.com/oai?verb=ListRecords&metadataPrefix=marcxml', 'list_records_with_namespaces.xml')
+        repository = described_class.new('http://www.example.com/oai')
+        record = repository.records(:metadata_prefix => 'marcxml').first
+
+        expect(record).to have_attributes(:identifier => '123456',
```

Comment:
do we want to try and test that it extracts headers as well?

Reply:
I think I covered all of them now.